backend/src/main/java/org/cryptomator/hub/api/AuthorityResource.java

```diff
@@ -0,0 +1,38 @@
+package org.cryptomator.hub.api;
+
+import org.cryptomator.hub.entities.Authority;
+import org.cryptomator.hub.entities.Group;
+import org.cryptomator.hub.entities.User;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.jboss.resteasy.annotations.cache.NoCache;
+import org.jboss.resteasy.annotations.jaxrs.QueryParam;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+@Path("/authorities")
+@Produces(MediaType.TEXT_PLAIN)
+public class AuthorityResource {
+
+	@GET
+	@Path("/search")
+	@RolesAllowed("user")
+	@Produces(MediaType.APPLICATION_JSON)
+	@NoCache
+	@Operation(summary = "search authority")
+	public List<AuthorityDto> search(@QueryParam("query") String query) {
+		return Authority.byName(query).map(authority -> {
+			if (authority instanceof User user) {
+				return new UserDto(authority.id, authority.name, user.pictureUrl, user.email, null);
+			} else if (authority instanceof Group) {
+				return new GroupDto(authority.id, authority.name);
+			}
+			throw new IllegalStateException("authority is not of type user or group");
```

Comment:
Can Authority be a sealed class, so the compiler knows that those two ifs are exhaustive?

Alternatively, I would prefer to put this throw statement into an else block for easier readability (you can see the branches are mutually exclusive without reading what happens inside the ifs)

Reply:
I get a runtime using a sealed class `Caused by: java.lang.IncompatibleClassChangeError: class org.cryptomator.hub.entities.Authority$HibernateProxy$qGonujMw cannot inherit from sealed class org.cryptomator.hub.entities.Authority` so I implemented the other variant in https://github.com/cryptomator/hub/pull/145/commits/af5a66aa5d2078716e3c79112eeec2838f4161e1